src/pgp.rs

```diff
@@ -246,25 +251,78 @@ pub fn pk_decrypt(
     let empty_pw = Password::empty();
 
     let decrypt_options = DecryptionOptions::new();
+    let try_symmetric_decryption = should_try_symmetric_decryption(&msg);
+    if try_symmetric_decryption.is_err() {
+        shared_secrets = &[];
+    }
+
+    // We always try out all passwords here, which is not great for performance.
+    // But benchmarking (see `benchmark_decrypting.rs`)
+    // showed that the performance impact is negligible.
+    // We could include a short (~2 character) identifier of the secret in cleartext
```

Comment:
There are other possible optimizations suggested (trying recently used secrets first, "To" aliases), i think we can omit mentioning them in the comments.